binding/web/src/types.ts

```diff
@@ -0,0 +1,113 @@
+/*
+  Copyright 2022 Picovoice Inc.
+
+  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+  file accompanying this source.
+
+  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+*/
+
+export type RhinoInitConfig = {
+  processErrorCallback?: (error: string) => void;
+  /** @defaultValue '0.5' */
+  sensitivity?: number;
+  /** @defaultValue '1.0' */
+  endpointDurationSec?: number;
+  /** @defaultValue 'false' */
+  requireEndpoint?: boolean;
+};
+
+export type RhinoInputConfig = {
+  /** @defaultValue 'rhino_model' */
+  customWritePath?: string;
+  /** @defaultValue false */
+  forceWrite?: boolean;
+  /** @defaultValue 1 */
+  version?: number;
+};
+
+export type RhinoOptions = RhinoInitConfig & RhinoInputConfig;
+
+export type RhinoContext = {
+  /** An arbitrary label used for caching purposes */
+  label: string;
+  /** Base64 representation of a trained Rhino context (`.rhn` file) */
+  base64?: string;
+  /** The Rhino context (`.rhn` file) path relative to the public directory */
+  rhnPath?: string;
```

Comment:
I wonder if we should call this just `filePath` or `publicPath` or something. Just seems weird to call it by the shorthand here.

Reply:
I mean it is the path to an `rhn`, I did it this way because in Porcupine it's the same, but `publicPath` might be a better choice since it mirrors the params interface better.

Reply:
yeah let's go with `publicPath`. Just makes it more clear overall what the variable represents in the API. We'll make sure porcupine does the same